File: 240418/최고의 13위치 2/best-place-of-13-2.py
N = int(input())
grid = [list(map(int, input().split())) for _ in range(N)]

answer = 0
for i in range(N):
    for j in range(N - 2):
        for k in range(N):
            for l in range(N - 2):
                if i == k and abs(l - j) <= 2:
                    continue

                cnt1 = grid[i][j] + grid[i][j + 1] + grid[i][j + 2]
                cnt2 = grid[k][l] + grid[k][l + 1] + grid[k][l + 2]
                answer = max(answer, cnt1 + cnt2)

print(answer)

File: 260215/최고의 13위치 2/best-place-of-13-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N;
    cin >> N;
    vector<vector<int>> grid(N, vector<int>(N));
    for (int i = 0; i < N; i++) 
    {
        for (int j = 0; j < N; j++) cin >> grid[i][j];
    }

    int answer = 0;

    for (int i = 0; i < N; i++) 
    {
        for (int j = 0; j <= N - 3; j++) 
        {
            for (int k = 0; k < N; k++) 
            {
                for (int l = 0; l <= N - 3; l++) 
                {

                    if (i == k && abs(l - j) <= 2) continue;

                    int cnt1 = grid[i][j] + grid[i][j + 1] + grid[i][j + 2];
                    int cnt2 = grid[k][l] + grid[k][l + 1] + grid[k][l + 2];
                    answer = max(answer, cnt1 + cnt2);
                }
            }
        }
    }

    cout << answer << "\n";
    return 0;
}
